lib/sign_in/idme/configuration.rb

```diff
@@ -71,7 +71,19 @@ def jwt_decode_algorithm
       end
 
       def jwt_decode_public_key
-        OpenSSL::PKey::RSA.new(File.read(Settings.idme.oauth_public_key))
+        IdmePublicKey.public_keys ||= fetch_idme_public_keys
+      end
+
+      def fetch_idme_public_keys
+        uri = URI(Settings.idme.oauth_public_key_url)
```

Comment:
I think this would be a good candidate for a service object class though: `IdmePublicKeyFetcher` or something like that